csrc/java_evp_keys.cpp

```diff
@@ -496,21 +496,19 @@ JNIEXPORT jlong JNICALL Java_com_amazon_corretto_crypto_provider_EvpKeyFactory_r
         if (privateExponentArr) {
             BigNumObj privExp = BigNumObj::fromJavaArray(env, privateExponentArr);
 
-            int res = 1;
             if (BN_is_zero(pubExp)) {
                 // RSA blinding can't be performed without |e|.
-                rsa.set(RSA_new_private_key_no_e(modulus, privExp));
+                rsa.set(new_private_RSA_key_with_no_e(modulus, privExp));
+                // new_private_RSA_key_with_no_e does not take the ownership of its arguments
             } else {
-                res = RSA_set0_key(rsa, modulus, pubExp, privExp);
-            }
-
-            if (res != 1) {
-                throw_openssl(EX_RUNTIME_CRYPTO, "Unable to set RSA values");
+                if (1 != RSA_set0_key(rsa, modulus, pubExp, privExp)) {
```

Comment:
nit: the convention is `function() != 1`

Reply:
Done.